openhtmltopdf-core/src/main/java/com/openhtmltopdf/context/StyleReference.java

```diff
@@ -251,22 +249,22 @@ private List getStylesheets() {
         StylesheetInfo[] refs = _nsh.getStylesheets(_doc);
         int inlineStyleCount = 0;
         if (refs != null) {
-            for (int i = 0; i < refs.length; i++) {
+            for (StylesheetInfo ref : refs) {
                 String uri;
-                
-                if (! refs[i].isInline()) {
-                    uri = _uac.resolveURI(refs[i].getUri());
-                    refs[i].setUri(uri);
+
+                if (!ref.isInline()) {
+                    uri = _uac.resolveURI(ref.getUri());
+                    ref.setUri(uri);
                 } else {
-                    refs[i].setUri(_uac.getBaseURL() + "#inline_style_" + (++inlineStyleCount));
+                    ref.setUri(_uac.getBaseURL() + "#inline_style_" + (++inlineStyleCount));
                     Stylesheet sheet = _stylesheetFactory.parse(
-                            new StringReader(refs[i].getContent()), refs[i]);
-                    refs[i].setStylesheet(sheet);
-                    refs[i].setUri(null);
+                            new StringReader(ref.getContent()), ref);
+                    ref.setStylesheet(sheet);
+                    ref.setUri(null);
                 }
             }
+            infos.addAll(Arrays.asList(refs));
```

Comment:
This is something IDEA cached ... if refs == null, the addAll would throw an NPE...